.github/workflows/ci.yml

```diff
@@ -5,16 +5,49 @@ on:
     branches:
       - "main"
 jobs:
-  tests:
+  linux:
+    strategy:
+      matrix:
+        system: [ "x86_64-linux", "aarch64-linux"]
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
-    - uses: cachix/install-nix-action@v25
+    - uses: docker/setup-qemu-action@v2
+    - uses: DeterminateSystems/nix-installer-action@main
       with:
-        github_access_token: ${{ secrets.GITHUB_TOKEN }}
+        extra-conf: |
+          accept-flake-config = true
+          extra-platforms = aarch64-linux
+    - uses: DeterminateSystems/magic-nix-cache-action@main
+      with: 
+        use-flakehub: false
     - uses: cachix/cachix-action@v14
       with:
         name: entropia
         authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
         extraPullNames: nix-community
-    - run: nix flake check
+        useDaemon: false
+    - run: nix run github:Mic92/nix-fast-build -- --skip-cached --no-nom --flake .#checks.${{ matrix.system }}
+
+  macos:
+    strategy:
+      matrix:
+        os: [ "macos-13", "macos-14" ]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v4
+    - uses: DeterminateSystems/nix-installer-action@main
+      with:
+        extra-conf: |
+          accept-flake-config = true
+          extra-platforms = aarch64-linux
```

Comment:
This seems wrong. Why should we have a Linux platform on macOS? Also not sure if we need that option at all (here and in Linux)

Reply:
That's a classical copy paste messup